src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketWebhookClientImpl.java

```diff
@@ -0,0 +1,46 @@
+package com.atlassian.bitbucket.jenkins.internal.client;
+
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketPage;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketWebhook;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketWebhookRequest;
+import com.fasterxml.jackson.core.type.TypeReference;
+import okhttp3.HttpUrl;
+
+import static java.util.Arrays.stream;
+
+public class BitbucketWebhookClientImpl implements BitbucketWebhookClient {
+
+    private final BitbucketRequestExecutor bitbucketRequestExecutor;
+    private final HttpUrl.Builder urlBuilder;
+
+    public BitbucketWebhookClientImpl(String projectKey,
+                                      String repoSlug,
+                                      BitbucketRequestExecutor bitbucketRequestExecutor) {
+        this.bitbucketRequestExecutor = bitbucketRequestExecutor;
+        this.urlBuilder = getWebhookUrl(projectKey, repoSlug);
```

Comment:
I don't think you should do this here. 
If I do
```java
//get my BitbucketWebhookClientImpl
bitbucketWebhookClientImpl.getWebhooks("one", "two);
bitbucketWebhookClientImpl.registerWebhookUrl(request);
```
the post request will have `?eventId=one&eventId=two` on it, right? I don't think we want that.